packages/taquito-michelson-encoder/src/tokens/pair.ts

```diff
@@ -169,11 +169,11 @@ export class PairToken extends ComparableToken {
     const leftToken = this.createToken(args[0], this.idx);
     let keyCount = 1;
     let leftValue;
-    if (leftToken instanceof PairToken) {
-      keyCount = Object.keys(leftToken.ExtractSchema()).length;
-    }
     if (leftToken instanceof PairToken && !leftToken.hasAnnotations()) {
       leftValue = getLeftValue(leftToken);
+      if (leftToken instanceof PairToken) {
```

Comment:
can you give a little explanation on why it's moved into the if block? for my own reference

Reply:
From Peter's document:

> The Pseudo code corresponding to the actual implementation differs. In managing the idx keyCount, it counts keys regardless of whether a nested pair has annotation or not.